src/node_crypto.cc

```diff
@@ -3530,6 +3535,9 @@ bool Hash::HashInit(const char* hash_type) {
     return false;
   EVP_MD_CTX_init(&mdctx_);
   EVP_DigestInit_ex(&mdctx_, md_, nullptr);
+  if (0 != ERR_peek_error()) {
+    return false;
+  }
```

Comment:
@ncopa Thank you for bringing this up. Let's continue the discussion under https://github.com/nodejs/node/issues/4221.